src/Bicep.Core/Emit/ParametersFileTemplateWriter.cs

```diff
@@ -0,0 +1,117 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using Bicep.Core.Semantics;
+using Bicep.Core.Syntax;
+using Microsoft.WindowsAzure.ResourceStack.Common.Json;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+using System;
+using System.IO;
+
+namespace Bicep.Core.Emit
+{
+    public class ParametersFileTemplateWriter : TemplateWriter
+    {
+        public ParametersFileTemplateWriter(SemanticModel semanticModel, EmitterSettings settings): base(semanticModel, settings)
+        {
+            this.context = new EmitterContext(semanticModel, settings);
+            this.settings = settings;
+        }
+
+        private readonly EmitterContext context;
+        private readonly EmitterSettings settings;
+
+        public void Write(JsonTextWriter writer, string existingContent)
+        {
+            var existingParamsContent = null as JToken;
+            var existingContentVersion = "1.0.0.0";
+            if (!string.IsNullOrWhiteSpace(existingContent))
+            {
+                try
+                {
+                    existingParamsContent = JToken.Parse(existingContent);
+                    existingContentVersion = existingParamsContent.Value<string>("contentVersion") ?? existingContentVersion;
+                }
+                catch
+                {
+                    // content of the existing file is not valid json, ignore merging it
```

Comment:
Thanks, but I think it would be better to throw an error.  It will be a surprise for the user to have their file overwritten if there are (unbeknownst them to) errors in it.

Reply:
P2